modules/videoio/src/videoio_registry.cpp

```diff
@@ -129,93 +135,94 @@ bool sortByPriority(const VideoBackendInfo &lhs, const VideoBackendInfo &rhs)
     return lhs.priority > rhs.priority;
 }
 
+inline static std::vector<std::string> tokenize_string(const std::string& input, char token = ',')
+{
+    std::vector<std::string> result;
+    std::string::size_type prev_pos = 0, pos = 0;
+    while((pos = input.find(token, pos)) != std::string::npos)
+    {
+        result.push_back(input.substr(prev_pos, pos-prev_pos));
+        prev_pos = ++pos;
+    }
+    result.push_back(input.substr(prev_pos));
+    return result;
+}
+
 /** @brief Manages list of enabled backends
  */
 class VideoBackendRegistry
 {
+public:
+    typedef std::vector<VideoBackendInfo> BackendsVec;
 protected:
-    std::vector<VideoBackendInfo> enabledBackends;
+    typedef std::map< VideoCaptureAPIs, Ptr<DynamicBackend> > PluginsMap;
+    BackendsVec enabledBackends;
+    PluginsMap plugins;
     VideoBackendRegistry()
     {
-        const int N = sizeof(builtin_backends)/sizeof(builtin_backends[0]);
-        enabledBackends.assign(builtin_backends, builtin_backends + N);
-        for (int i = 0; i < N; i++)
+        typedef std::vector<std::string> PriorityVec;
+        using namespace cv::utils;
+        const std::string backendOrder_str = getConfigurationParameterString("OPENCV_VIDEOIO_PRIORITY_LIST", NULL);
+        const PriorityVec backendOrder = tokenize_string(backendOrder_str);
+        if (!backendOrder.empty())
         {
-            VideoBackendInfo& info = enabledBackends[i];
-            info.priority = 1000 - i * 10;
+            CV_LOG_INFO(NULL, "VIDEOIO: Configured priority list (OPENCV_VIDEOIO_PRIORITY_LIST): " << backendOrder_str);
         }
-        CV_LOG_DEBUG(NULL, "VIDEOIO: Builtin backends(" << N << "): " << dumpBackends());
-        if (readPrioritySettings())
-        {
-            CV_LOG_INFO(NULL, "VIDEOIO: Updated backends priorities: " << dumpBackends());
-        }
-        int enabled = 0;
-        for (int i = 0; i < N; i++)
+
+        const int N = sizeof(builtin_backends)/sizeof(builtin_backends[0]);
+        for (int i = 0; i < N; ++i)
         {
-            VideoBackendInfo& info = enabledBackends[enabled];
-            if (enabled != i)
-                info = enabledBackends[i];
-            size_t param_priority = utils::getConfigurationParameterSizeT(cv::format("OPENCV_VIDEOIO_PRIORITY_%s", info.name).c_str(), (size_t)info.priority);
-            CV_Assert(param_priority == (size_t)(int)param_priority); // overflow check
-            if (param_priority > 0)
+            VideoBackendInfo be = builtin_backends[i];
+
+            // Check if backend needs plugin
+            if (be.mode & MODE_DYNAMIC)
             {
-                info.priority = (int)param_priority;
-                enabled++;
+                Ptr<DynamicBackend> plugin = DynamicBackend::load(be.id, (int)be.mode);
```

Comment:
We should not load plugins here.
Lets just assign priorities (keep old code).

At least, there is no way to bypass plugin loading via `OPENCV_VIDEOIO_PRIORITY_XXX=0`.